about/async/suggest.js

```diff
@@ -28,32 +31,65 @@ exports.create = function (api) {
         })
       }
     }
-  })
+  }
 
   function loadSuggestions () {
-    if (!suggestions) {
-      var id = api.keys.sync.id()
-      var following = api.contact.obs.following(id)
-      var recentlyUpdated = api.feed.obs.recent()
-      var contacts = computed([following, recentlyUpdated], function (a, b) {
-        var result = Array.from(a)
-        b.forEach((item, i) => {
-          if (!result.includes(item)) {
-            result.push(item)
-          }
-        })
-        return result
+    if (suggestions) return
+
+    var id = api.keys.sync.id()
+    var following = api.contact.obs.following(id)
+    var recentlyUpdated = api.feed.obs.recent()
+    var contacts = computed([following, recentlyUpdated], (a, b) => {
+      var result = Array.from(a)
+      b.forEach(item => {
+        if (!result.includes(item)) {
+          result.push(item)
+        }
       })
+      return result
+    })
 
-      recentSuggestions = map(
-        computed(recentlyUpdated, (items) => Array.from(items).slice(0, 10)),
-        suggestion,
-        {idle: true}
-      )
-      suggestions = map(contacts, suggestion, {idle: true})
-      watch(recentSuggestions)
-      watch(suggestions)
-    }
+    recentSuggestions = map(
+      computed(recentlyUpdated, (items) => Array.from(items).slice(0, 10)),
+      suggestion,
+      {idle: true}
+    )
+
+    const suggestionsRecord = lookup(contacts, contact => {
+      return [contact, api.about.obs.names(contact)]
+    })
+
+    suggestions = concat(
+      map(dictToCollection(suggestionsRecord), pluralSuggestions, {idle: true})
+    )
+
+    watch(recentSuggestions)
+    watch(suggestions)
+  }
+
+
+  function pluralSuggestions (item) {
```

Comment:
@mixmix I think this could be better, but if it works and isn't slow, probably okay. Will have a proper look tomorrow and see what I can come up with.